docs/api.md

```diff
@@ -7,102 +7,167 @@ method arguments and return values.
+originally appeared in the 2014 Mid-Central ICPC. It was first used in an AUACM competition with id 6. The
+problem has no description, input description, output description, or sample cases. It is named (More) Multiplication,
+with a problem id of 34, and is also uniquely identified by the string "multiplication". (More) Multiplication
+has a difficulty of 67.
+
+### The Sample Case Object
+
+| Name | Type | Description |
+| --- | --- | --- |
+|`input`|`String`|The input fed to standard in for a test run|
+|`output`|`String`|The output that would be produced to standard out by a correct program for that test run|
+
+__Example:__
+
+```json
+{
     "input": "1",
     "output": "2"
-  },
-  {
-    "input": "2",
-    "output": "3"
-  }
-]
+}
 ```
 
-If any of the required fields are not supplied, the API will return with status
-code 400. If successful, the API will return a JSON object representation of
-the newly created problem.
+The above sample cases object represents a sample case where an input of "1" to a program that
+correctly solves the associated problem produces a corresponding output of "2".
+
+## Problem Management Endpoints
+
+### Get Data on all problems
+
+__URL:__ `/api/problems`
+
+__Method:__ `GET`
+
+__Returns:__ a JSON array of all the publicly available data on all the problems
+in the database.
+Specifically, the method returns an array of [problem objects](#the-problem-object)
+describing every public problem in the database.
+
+__Example use:__
 
+`GET /api/problems`
 
 ### Get Data on a Problem
 
 __URL:__ `/api/problems/{identifier}`
 
- _Note:_ A problem's identifier can be its numeric id (i.e. 1) or it's
+ _Note:_ A problem's identifier can be its numeric id (i.e. 1) or its
  alphanumeric "shortname" (i.e. blackvienna)
 
 __Method:__ `GET`
 
-Returns detailed data on a specific problem as a JSON object.
-Specifically, the method returns
-  * __Problem ID__
-  * __Full Name__
-  * __Short Name__
-  * __Competition Appearance__
-  * __Difficulty Rating__
-  * __AUACM Competition Release__
-  * __Description__
-  * __Input Description__
-  * __Output Description__
-  * __Sample Cases Array__
+__Returns:__ detailed data on a specific problem as a JSON object.
+Specifically, the method returns a [problem object](#the-problem-object).
 
+__Example uses:__
 
-### Get Data on all problems
+`GET /api/problems/multiplication`
 
-__URL:__ `/api/problems/`
+`GET /api/problems/34`
 
-__Method:__ `GET`
 
-Returns a JSON array of all the publicly available data on all the problems problems
-in the database.
-Specifically, the method returns
-  * __pid__
-  * __full name__
-  * __short name__
-  * __competition appearance__
-  * __difficulty rating__
-  * __auacm competition release__
-  * __date added__
-  * __solved by user__
-  * __problem url__
-
-of all the problems.
+### Create a New Problem
+***This method requires being logged in as an administrator***
```

Comment:
I'd prefer "This endpoint..." over "This method...".

Also, consider grouping all of the admin-only endpoints into one section.